src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunctionSet.cs

```diff
@@ -0,0 +1,273 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Functions
+{
+    internal class TexlFunctionSet<T>
+        where T : ITexlFunction
```

Comment:
>T : ITexlFunction [](http://example.com/codeflow?start=14&length=17)

Why do we need this interface/generic param? When will this not be the concrete `TexlFunction`?

Reply:
We're in Core and can't access Interpreter with AsyncFunctionPtr

Reply:
Used in Interpreter, in TexlFunctionImplementation and Library.cs